src/common/com/intellij/plugins/haxe/lang/psi/HaxeClassResolveResult.java

```diff
@@ -152,23 +183,66 @@ public void specialize(@Nullable PsiElement element) {
     }
   }
 
+  /**
+   * Creates a resolved parameter list that can be used with specializedByParameters() when resolving
+   * subclasses.
+   *
+   * @param targetParam - the haxe class that is being resolved
+   * @param innerSpecialization - the specializationlist
+   * @return
+   */
+  @Nullable
+  private static List<PsiElement> generateParameterList(HaxeTypeParam targetParam, HaxeGenericSpecialization innerSpecialization) {
+    HaxeTypeList typeList = targetParam == null ? null : targetParam.getTypeList();
+    if (null == typeList) {
+      return null;
+    }
+    List<PsiElement> instantiationParams = new ArrayList<PsiElement>();
+    for (HaxeTypeListPart part : typeList.getTypeListPartList()) {
+      HaxeType type = getTypeOfTypeListPart(part);
+      final String name = type != null ? type.getText() : null;
+
+      HaxeClassResolveResult resolvedParam = innerSpecialization.get(null, name);
+      HaxeClass resolvedClass = null != resolvedParam ? resolvedParam.getHaxeClass() : null;
+      if (null == resolvedClass) {
+        resolvedParam = HaxeResolveUtil.getHaxeClassResolveResult(type); // No specialization??
+        resolvedClass = null != resolvedParam ? resolvedParam.getHaxeClass() : null;
+      }
+      instantiationParams.add(resolvedClass);
+    }
+    return instantiationParams;
+  }
+
   public void specializeByParameters(@Nullable HaxeTypeParam param) {
     if (param == null || haxeClass == null || !haxeClass.isGeneric()) {
       return;
     }
+    List<PsiElement> specializedTypes = new ArrayList<PsiElement>();
+    final HaxeTypeList typeList = param.getTypeList();
+    for (int i = 0; i < typeList.getTypeListPartList().size(); i++) {
+      final HaxeType specializedType = getTypeOfTypeListPart(typeList.getTypeListPartList().get(i));
+      specializedTypes.add(specializedType);  // OK to be null
+    }
+    specializeByParameters(specializedTypes);
+  }
+
+  public void specializeByParameters(@Nullable List<PsiElement> typeList) {
```

Comment:
I had to break specializeByParameters into two pieces so that I could use it for real types in a new list as well as keep the old interface, which is used in a couple of other places.